src/transformers/models/sam/modeling_sam.py

```diff
@@ -0,0 +1,1337 @@
+            module.weight.data.normal_(mean=0.0, std=std)
+            if module.padding_idx is not None:
+                module.weight.data[module.padding_idx].zero_()
+
+
+SAM_START_DOCSTRING = r"""
+    This model inherits from [`PreTrainedModel`]. Check the superclass documentation for the generic methods the
+    library implements for all its model (such as downloading or saving, resizing the input embeddings, pruning heads
+    etc.)
+
+    This model is also a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass.
+    Use it as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage
+    and behavior.
+
+    Parameters:
+        config ([`SamConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+
+SAM_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`SamProcessor`]. See [`SamProcessor.__call__`] for
+            details.
+        input_points (`torch.FloatTensor` of shape `(batch_size, num_points, 2)`):
+            Input 2D spatial points, this is used by the prompt encoder to encode the prompt. Generally yields to much
+            better results. The points can be obtained by passing a list of list of list, corresponding to the batch,
+            the number of points per image and the x and y coordinate of the point. If a batch of images is passed,
+            "PAD" points are automatically generated and ignored by the prompt encoder.
+        input_labels (`torch.LongTensor` of shape `(batch_size, num_points)`):
+            Input labels for the points, this is used by the prompt encoder to encode the prompt. Generally yields to
+            according to the official implementation, there are 3 types of labels
+            - `1`: the point is a point that contains the object of interest
+            - `0`: the point is a point that does not contain the object of interest
+            - `-1`: the point corresponds to the background
+            We added the label:
+            - `-10`: the point is a padding point, thus should be ignored by the prompt encoder
+            The padding of the point is automatically done by the processor.
+        input_boxes (`torch.FloatTensor` of shape `(batch_size, num_boxes, 4)`):
+            Input boxes for the points, this is used by the prompt encoder to encode the prompt. Generally yields to
+            much better results. The boxes can be obtained by passing a list of list of list, corresponding to the
+            batch, the number of boxes per image and the x1, y1, x2, y2 coordinates of the box.
```

Comment:
What are x1, y1, x2 and y2? Be explicit.
Same comment as before on the list of list of list.

Here in the documentation of the model, explain clearly what the model expects.